core/src/main/java/org/testcontainers/containers/GenericContainer.java

```diff
@@ -359,17 +372,23 @@ private void tryStart(Instant startedAt) {
 
                         createCommand.getLabels().put(HASH_LABEL, hash);
                     }
+                    reusable = true;
                 } else {
                     logger().info("Reuse was requested but the environment does not support the reuse of containers");
+                    reusable = false;
                 }
             } else {
+                reusable = false;
+            }
+
+            if (!reusable) {
                 createCommand.getLabels().put(DockerClientFactory.TESTCONTAINERS_SESSION_ID_LABEL, DockerClientFactory.SESSION_ID);
             }
 
             if (!reused) {
                 containerId = createCommand.exec().getId();
 
-                // TODO add to the hash
+                // TODO use single "copy" invocation (and calculate an hash of the resulting tar archive)
```

Comment:
What's the driving force to do this? Efficiency, or something else?

Reply:
Efficiency, yes. Especially if we can calculate the hash as part of the same operation

Does not mean that the current approach is slow, just a nice TODO to remember